docs/cicd/others-script-cd-template.sh

```diff
@@ -0,0 +1,64 @@
+# This is just an example workflow for continuous deployment.
+# You should customize it to meet your own requirements.
+# Also you should export the following environment variables for Azure/M365 login:
+# AZURE_ACCOUNT_NAME
+# AZURE_ACCOUNT_PASSWORD
+# AZURE_SUBSCRIPTION_ID
+# AZURE_TENANT_ID
+# M365_ACCOUNT_NAME
+# M365_ACCOUNT_PASSWORD
+
+# Setup environment.
+# Sufficient permissions are required to run the commands below.
+apt install -y nodejs npm git
+
+# Checkout the code.
+git clone {RepositoryEndpoint}
+cd {FolderName}
+
+# Install the local dev dependency of @microsoft/teamsfx-cli. 
+# 'npm ci' is used here to install dependencies and it depends on package-lock.json.
+# If you prefer to use 'npm ci', please make sure to commit package-lock.json first, or just change it to 'npm install'.
+npm ci
+
+# Build the project.
+# The way to build the current project depends on how you scaffold it.
+# Different folder structures require different commands set.
+cd tabs && npm ci && npm run build
+
+# Run unit test.
+# Currently, no opinioned solution for unit test provided during scaffolding, so,
+# set up any unit test framework you prefer (for example, mocha or jest) and update the commands accordingly in below.
+npm run test
+
+# We suggest to do the provision steps by case manually or in a separated workflow, so just comment the following steps for references.
+# After provision, you should commit .fx/env.default.json into the repository for later use.
+# You should pick required secrets from .fx/default.userdata, and export them in your environment which can be refered by the step with name 'Generate default.userdata'. 
+
+# Provision hosting environment.
+# npx teamsfx provision --subscription ${AZURE_SUBSCRIPTION_ID}
+
+# Commit provision configs if necessary.
+# git add .fx/env.default.json
+# git commit -m "chore: commit provision configs"
+# git push
+
+# Generate default.userdata
+[ ! -z "${USERDATA_TENANT_ID}" ] && echo "solution.teamsAppTenantId=${USERDATA_TENANT_ID}" >> .fx/default.userdata
+[ ! -z "${USERDATA_CLIENT_SECRET}" ] && echo "fx-resource-aad-app-for-teams.clientSecret=${USERDATA_CLIENT_SECRET}" >> .fx/default.userdata
+[ ! -z "${USERDATA_BOT_PASSWORD}" ] && echo "fx-resource-bot.botPassword=${USERDATA_BOT_PASSWORD}" >> .fx/default.userdata
+
+# Deploy to hosting environment.
+cd .. && npx teamsfx deploy
```

Comment:
this is ok as is, but we technically shouldn't have to refer to teamsfx here, this can be an npm script `"deploy" : "teamsfx deploy"` and this code here can become agnostic of teamsfx `npm run deploy` this enables more flexibility for developers where they can tweak their deploy/build scripts without having to change CI scripts. This is fine as is for now though.

Reply:
Good ideas to decouple deployment details to a specific teamsfx command. Let's keep it as it is first.